src/components/flat-table/flat-table-row/flat-table-row.style.js

```diff
@@ -49,33 +50,55 @@ const StyledFlatTableRow = styled.tr`
     }
   `}
 
-  ${({ selected, highlighted, theme }) => css`
+  ${({
+    selected, highlighted, isInSidebar, theme
+  }) => css`
+
+    ${isInSidebar && `
+      ${StyledFlatTableHeader},
+      ${StyledFlatTableRowHeader},
+      ${StyledFlatTableCell},
+      ${StyledFlatTableCheckbox} {
+        background-color: ${theme.flatTable.drawerSidebar.headerBackground};
+      }
+      
+        :hover {
+          ${StyledFlatTableCell}, ${StyledFlatTableCheckbox}:not(th) {
+            background-color: ${theme.flatTable.drawerSidebar.hover};
+          }
+        }
+
+        ${StyledFlatTableCheckbox}:not(th) {
+          border-right: 1px solid ${theme.flatTable.drawerSidebar.hover};
+        }
+    `}
+
     ${highlighted && `
       ${StyledFlatTableCell}, ${StyledFlatTableCheckbox} {
-        background-color: ${theme.flatTable.highlighted};
-        border-bottom-color: ${theme.flatTable.highlighted};
+        background-color: ${isInSidebar ? theme.flatTable.drawerSidebar.highlighted : theme.flatTable.highlighted};
+        border-bottom-color: ${isInSidebar ? theme.flatTable.drawerSidebar.highlighted : theme.flatTable.highlighted};
       }
 
       :hover {
         ${StyledFlatTableCell},
         ${StyledFlatTableRowHeader},
-        ${StyledFlatTableCheckbox} {
-          background-color: ${theme.flatTable.highlighted};
+        ${StyledFlatTableCheckbox}:not(th) {
+          background-color: ${isInSidebar ? theme.flatTable.drawerSidebar.highlighted : theme.flatTable.highlighted};
         }
       }
     `}
 
     ${selected && `
       ${StyledFlatTableCell}, ${StyledFlatTableCheckbox} {
-        background-color: ${theme.flatTable.selected};
-        border-bottom-color: ${theme.flatTable.selected};
+        background-color: ${isInSidebar ? theme.flatTable.drawerSidebar.selected : theme.flatTable.selected};
+        border-bottom-color: ${isInSidebar ? theme.flatTable.drawerSidebar.selected : theme.flatTable.selected};
       }
 
       :hover {
         ${StyledFlatTableCell},
         ${StyledFlatTableRowHeader},
-        ${StyledFlatTableCheckbox} {
-          background-color: ${theme.flatTable.selected};
+        ${StyledFlatTableCheckbox}:not(th) {
+          background-color: ${isInSidebar ? theme.flatTable.drawerSidebar.selected : theme.flatTable.selected};
```

Comment:
`${isInSidebar ? theme.flatTable.drawerSidebar.selected : theme.flatTable.selected};`
this is repeated a lot of times, it could be set as a const.

Reply:
Fair point I'll add that in